CMakeLists.txt

```diff
@@ -102,49 +102,26 @@ else()
-# APP_VERSION and APP_TARGET defines in the source code if needed.
-#
-# APP_BUILD is going away (it is not currently used in the source code now as
-# Mudlet instead reads it from the resource file) however until the CI/CB system
-# is cleaned up to not use it in any way in the
-# /CI/travis.validate_deployment.sh script we probably have to leave it in
-# place:
-#
-# IMPORTANT: To insure consistency please ensure the SAME of the first two
-# values are also assigned to the "VERSION" and "BUILD" variables in the native
-# qmake project file, which is NOW called: ./src/mudlet.pro
-# NOTE: the CI/CB build process examines the following line to determine the
-# underlying version of the Mudlet build for all our current CI/CB processes
-# even a Windows one that currently runs using qmake - so don't mess with it
-# unless you are a Core Developer adjusting the overall versioning:
+# Set APP_VERSION (update in mudlet.pro as well)
 set(APP_VERSION 4.17.2)
-# We now make APP_BUILD a CACHE variable so it can be propagated to the ./src/
-# sub-directory build.
-if(DEFINED ENV{MUDLET_VERSION_BUILD} AND NOT $ENV{MUDLET_VERSION_BUILD}
-                                         STREQUAL "")
-  file(WRITE ${CMAKE_SOURCE_DIR}/src/app-build.txt "$ENV{MUDLET_VERSION_BUILD}-${GIT_SHA1}")
-  message(STATUS "Value written to app-build.txt file: $ENV{MUDLET_VERSION_BUILD}-${GIT_SHA1}")
-  set(APP_BUILD $ENV{MUDLET_VERSION_BUILD} CACHE STRING "This variable is automatically set during the CMake run and should not be manually altered!" FORCE)
+
+# Set APP_BUILD based on environment variable MUDLET_VERSION_BUILD or default
+if(DEFINED ENV{MUDLET_VERSION_BUILD} AND NOT $ENV{MUDLET_VERSION_BUILD} STREQUAL "")
+    set(BUILD_VALUE "$ENV{MUDLET_VERSION_BUILD}-${GIT_SHA1}")
 else()
-  file(WRITE ${CMAKE_SOURCE_DIR}/src/app-build.txt "-dev-${GIT_SHA1}")
-  message(STATUS "Value written to app-build.txt file: -dev-${GIT_SHA1}")
-  set(APP_BUILD "-dev" CACHE STRING "This variable is automatically set during the CMake run and should not be manually altered!" FORCE)
-  # Core dev team setting things up for a release should comment out the above
-  # three lines and uncomment these below - note that the app-build.txt must not
-  # contain anything (other than whitespace) for a RELEASE build:
-  #file(WRITE ${CMAKE_SOURCE_DIR}/src/app-build.txt " ")
-  #message(STATUS "Value written to app-build.txt file: {nothing}")
-  #set(APP_BUILD "" CACHE STRING "This variable is automatically set during the CMake run and should not be manually altered!" FORCE)
+    set(BUILD_VALUE "-dev-${GIT_SHA1}")
 endif()
 
+# Write to app-build.txt and set APP_BUILD
+file(WRITE ${CMAKE_SOURCE_DIR}/src/app-build.txt "${BUILD_VALUE}")
+message(STATUS "Value written to app-build.txt file: ${BUILD_VALUE}")
+set(APP_BUILD ${BUILD_VALUE} CACHE STRING "Auto-set during CMake run. Do not alter manually." FORCE)
+
+# For release builds, comment out the above and uncomment below:
```

Comment:
Maybe we should be explicit and mention THREE lines here?

Reply:
It would be better to change to a setup that doesn't require manually adjusting the file instead